src/main/java/actions/RunCodeProfilerAction.java

```diff
@@ -16,6 +21,66 @@ public RunCodeProfilerAction(CodeProfilerToolWindow codeProfiler) {
         this.codeProfiler = codeProfiler;
     }
 
+    /**
+     * @param hostname String: hostname used to get staging IP address from
+     * @return InetAddress: IP address of the staging hostname
+     * @throws NamingException      thrown if the CNAME lookup for the hostname fails
+     * @throws UnknownHostException thrown if the CNAME IP address cannot be resolved
+     */
+    private InetAddress getStagingIp(String hostname) throws NamingException, UnknownHostException {
+        if (hostname.contains("://")) {
+            // remove protocol from hostname
+            hostname = hostname.split("(://)")[1];
+        }
+
+        // Check if local url
+        String[] localAddresses = new String[]{
+                "127.0.0.1",
+                "0.0.0.0",
+                "localhost"
+        };
+        for (String address : localAddresses) {
+            if (hostname.equals(address)) {
+                return InetAddress.getLoopbackAddress();
+            }
+        }
+
+        // Get CNAME
+        String[] edgeHostEndings = new String[]{
+                ".edgekey.net",
+                ".edgesuite.net",
+                ".akamaiedge.net",
+                ".akamaized.net"
+        };
+        String[] edgeHostStagingEndings = new String[]{
+                ".edgekey-staging.net",
+                ".edgesuite-staging.net",
+                ".akamaiedge-staging.net",
+                ".akamaized-staging.net"
+        };
+        String cname = null;
+        for (int i = 0; i < 4; i++) {
+            if (hostname.endsWith(edgeHostEndings[i]) || hostname.endsWith(edgeHostStagingEndings[i])) {
+                // hostname is already the cname
+                cname = hostname;
+                break;
+            }
+        }
+        if (cname == null) {
+            // get cname if we don't already have it
+            cname = DnsService.getCNAME(hostname)[0];
```

Comment:
@Can you confirm for example for `www.ikea.com` which of that the CNAME comes back as the first result for a dig:

```
macbook:~$ dig www.ikea.com

; <<>> DiG 9.10.6 <<>> www.ikea.com
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 16366
;; flags: qr rd ra; QUERY: 1, ANSWER: 3, AUTHORITY: 0, ADDITIONAL: 2

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 512
;; QUESTION SECTION:
;www.ikea.com.			IN	A

;; ANSWER SECTION:
www.ikea.com.		86400	IN	CNAME	san.ov11632.ikea.com.edgekey.net.
san.ov11632.ikea.com.edgekey.net. 21600	IN CNAME e11632.x.akamaiedge.net.
e11632.x.akamaiedge.net. 20	IN	A	23.223.213.212

;; ADDITIONAL SECTION:
www.ikea.com.		1	IN	TXT	"ETPA"

;; Query time: 563 msec
;; SERVER: 127.0.0.1#53(127.0.0.1)
;; WHEN: Tue Nov 08 09:05:10 EST 2022
;; MSG SIZE  rcvd: 166

macbook:~$ 
```

In this case it's `san.ov11632.ikea.com.edgekey.net` -- does the lookup here use that or does it resolve down to the akamaiedge map?

Reply:
Yep, can confirm the code returns `san.ov11632.ikea.com.edgekey-staging.net`. I believe this is using the system DNS so in theory it should be the same as dig for everything.

Reply:
OK great I was worried the order might be different in the array that gets returned.